### PR TITLE
testing: fix TestRekeyUpgrade when round=0

### PR DESCRIPTION
**Pull request by tsachiherman**
## Summary

The previous fix for this test (#2178) was incomplete. It did not included proper handling for the case of round 0.
This PR addresses this corner case.

## Test Plan

This is a test.

### CODE REVIEW

File: test/e2e-go/upgrades/rekey_support_test.go [hunk 1]
```diff
@@ -63,8 +63,8 @@ func TestRekeyUpgrade(t *testing.T) {
 	initialStatus := curStatus
 	round := curStatus.LastRound
 
-	if initialStatus.NextVersion != initialStatus.LastVersion {
-		// no consensus upgrade took place
+	if initialStatus.NextVersion != initialStatus.LastVersion || round == 0 {
+		// no consensus upgrade took place (yet)
```
[the remark this answers is not in the document]
Comment (tsachiherman) [reply test/e2e-go/upgrades/rekey_support_test.go:67]: it could - and the code (should) handle that correctly.

File: test/e2e-go/upgrades/rekey_support_test.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/protocol"
 	"github.com/algorand/go-algorand/test/framework/fixtures"
 )
```
Comment (algonautshant) [inline test/e2e-go/upgrades/rekey_support_test.go:31]: Typo in the comment below: 
tests that **we** rekey -> tests that **the** rekey

**Review by algonautshant — APPROVED** [review]
Looks great.

File: test/e2e-go/upgrades/rekey_support_test.go (hunk 2 above)
Comment (tsachiherman) [reply test/e2e-go/upgrades/rekey_support_test.go:31]: I'll make that change before merging the code in if it would pass travis..
